app-ng/src/handlers.lisp

```diff
@@ -0,0 +1,359 @@
+;;;; app-ng/src/handlers.lisp
+;;;;
+;;;; author: appleby
+(in-package #:qvm-app-ng)
+
+(defvar *rpc-handlers* '()
+  "An alist of (rpc-method-name . handler-function) conses defined by DEFINE-RPC-HANDLER.")
+
+(defun lookup-rpc-handler-for-request (request-params)
+  (cdr (assoc (gethash "type" request-params) *rpc-handlers* :test #'string=)))
```

Comment:
Probably no biggie for an association list of ~14 items but it might be worth it to favor a `gethash` over a `cdr`+`assoc` as this seems to be run for each request.

Reply:
Fair point. Switching to a hash-table would also mean I don't need to do anything special to accommodate handler redefinitions.  For small tables like this I just instinctively prefer lists because the list-manipulating functions are more pleasant to work with (imo) than the hash-table APIs. So far I have tried to resist (unsuccessfully, see the "safety-hash" package :-D) small optimizations like this because of an untested assumption that in realistic access patterns for qvm-app, qvm allocation and simulation time will tend to dominate.

I've actually been coming around to the idea recently of just doing away with this global `*rpc-handlers*` table and possibly going back to an explicit dispatch function + caller-provided table a-la `rpcq:start-server` or indeed the handler dispatch in the existing `qvm-app`.  It's slightly less convenient from the standpoint of defining new handlers, but how often are you really defining new handlers in a SLIME REPL anyway?  So far I have resisted making the change in this PR in order to not create additional churn or add more lines of diff to an already oversized and long-in-tooth PR.

Reply:
I never could resist a juicy micro-optimization... I switched to a hash-table in the latest commit.